packages/editor/src/components/blocks/fieldset/Fieldset.tsx

```diff
@@ -0,0 +1,61 @@
+import type { Fieldset, Prettify } from '@axonivy/form-editor-protocol';
+import { DEFAULT_QUICK_ACTIONS, type ComponentConfig, type UiComponentProps } from '../../../types/config';
+import { ComponentBlock } from '../../../editor/canvas/ComponentBlock';
+import IconSvg from './Fieldset.svg?react';
+import { defaultBaseComponent, baseComponentFields } from '../base';
+import { EmtpyBlock } from '../../../editor/canvas/EmptyBlock';
+import { useAppContext } from '../../../context/AppContext';
+import { FIELDSET_DROPZONE_ID_PREFIX } from '../../../data/data';
+import './Fieldset.css';
+
+type FieldsetProps = Prettify<Fieldset>;
+
+export const defaultFieldsetProps: FieldsetProps = {
+  components: [],
+  legend: 'fieldset title',
+  disabled: false,
+  collapsible: false,
+  defaultCollapsed: false,
+  ...defaultBaseComponent
+};
+
+export const FieldsetComponent: ComponentConfig<FieldsetProps> = {
+  name: 'Fieldset',
+  category: 'Structure',
+  subcategory: 'General',
+  icon: <IconSvg />,
+  description: 'A group of inputs',
+  defaultProps: defaultFieldsetProps,
+  quickActions: DEFAULT_QUICK_ACTIONS,
+  render: props => <UiBlock {...props} />,
+  create: ({ defaultProps }) => ({ ...defaultFieldsetProps, ...defaultProps }),
+  outlineInfo: component => component.legend,
+  fields: {
+    components: { subsection: 'General', type: 'hidden' },
+    legend: { subsection: 'General', label: 'legend', type: 'textBrowser', browsers: ['ATTRIBUTE'] },
+    collapsible: { subsection: 'Behaviour', label: 'collapsible', type: 'checkbox' },
+    defaultCollapsed: { subsection: 'Behaviour', label: 'default collapsed', type: 'checkbox' },
+    disabled: { subsection: 'Behaviour', label: 'read-only', type: 'checkbox' },
+    ...baseComponentFields
+  }
+};
+
+const UiBlock = ({ id, components, legend, collapsible, disabled }: UiComponentProps<FieldsetProps>) => {
+  const { ui } = useAppContext();
+  return (
+    <>
+      <fieldset data-responsive-mode={ui.deviceMode} className={`${collapsible ? 'collapsible' : ''}`} disabled={disabled}>
+        <legend>{legend}</legend>
+        {components.map((component, index) => (
+          <ComponentBlock key={component.id} component={component} preId={components[index - 1]?.id} />
+        ))}
+        <EmtpyBlock
+          id={`${FIELDSET_DROPZONE_ID_PREFIX}${id}`}
+          preId={components[components.length - 1]?.id}
+          forLayout={true}
+          dragHint={{ display: components.length === 0, message: 'Drag first element inside the fieldset', mode: 'row' }}
+        />
+      </fieldset>
+    </>
+  );
+};
```

Comment:
I think you can simplify this a bit, as I think you don't need to set this data-responsive-mode attribute. 
This is used by the layout to simulate the device mode, but the fieldset does not make any layouting...

```suggestion
const UiBlock = ({ id, components, legend, collapsible, disabled }: UiComponentProps<FieldsetProps>) => (
  <fieldset className={`${collapsible ? 'collapsible' : ''}`} disabled={disabled}>
    <legend>{legend}</legend>
    {components.map((component, index) => (
      <ComponentBlock key={component.id} component={component} preId={components[index - 1]?.id} />
    ))}
    <EmtpyBlock
      id={`${FIELDSET_DROPZONE_ID_PREFIX}${id}`}
      preId={components[components.length - 1]?.id}
      forLayout={true}
      dragHint={{ display: components.length === 0, message: 'Drag first element inside the fieldset', mode: 'row' }}
    />
  </fieldset>
);
```